frontend/src/__tests__/cypress/cypress/pages/pipelines/pipelinesTable.ts

```diff
@@ -38,9 +46,12 @@ class PipelinesTable {
     return this.find().find('thead').findByRole('button', { name });
   }
 
-  getRowByName(name: string) {
-    return new PipelinesTableRow(() =>
-      this.find().findByTestId(`table-row-title-${name}`).parents('tr'),
+  getRowById(id: string) {
+    return new PipelinesTableRow(
+      () =>
+        this.find().findByTestId(['pipeline-row', id]) as unknown as Cypress.Chainable<
```

Comment:
Ideally we should not have to cast anything here.

Reply:
Can we log a task to look into this. I don't want to hold up the PR because react testing library cypress integration doesn't support better generic typing.

Reply:
Sure, if Juntao wants to look at this afterwards, that's fine with me.

Reply:
@DaoDaoNoCode please log a task for this so it doesn't get lost.

Reply:
https://issues.redhat.com/browse/RHOAIENG-7402 Created a ticket here.